rpcd/playbooks/roles/rpc_maas/tasks/maas_exclude.yml

```diff
@@ -13,18 +13,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-- name: Ensure local checks are removed
-  file:
-    path: "/etc/rackspace-monitoring-agent.conf.d/{{ item }}--{{ inventory_hostname }}.yaml"
-    state: absent
-  with_items:
-    - "{{ maas_excluded_checks }}"
-  delegate_to: "{{ physical_host }}"
+- name: Get a list of all checks that exist on the system
+  find:
+    paths: /etc/rackspace-monitoring-agent.conf.d/
+    patterns: "*.yaml"
+  register: maas_check_files
 
-- name: Ensure remote checks are removed
+- name: Ensure excluded checks are removed
   file:
-    path: "/etc/rackspace-monitoring-agent.conf.d/{{ item }}.yaml"
+    path: "{{ item.path }}"
     state: absent
   with_items:
-    - "{{ maas_excluded_checks }}"
+    - "{{ maas_check_files.files }}"
+  when:
+    - item.path | basename | match(maas_excluded_checks_regex)
```

Comment:
Not sure if it's of concern, but this match is actually matching against a filename, which may not be what people expect (since it's including `.yaml`).  I _think_ that's differing behaviour to how this worked before.